src/GitHub.App/Services/RepositoryCloneService.cs

```diff
@@ -103,6 +106,24 @@ public async Task<ViewerRepositoriesModel> ReadViewerRepositories(HostAddress ad
             return result;
         }
 
+        /// <inheritdoc/>
```

Comment:
I'm wondering whether we need both `Clone` and `CloneOrOpen`? Wouldn't `Clone` and `Open` make more sense? The decision of which to call would typically be made at a higher level I would have thought as we're checking whether the repository already exists in order to show/hide the `Open` button.

Reply:
I've updated this as discussed in Slack. There is now a `CloneOrOpenRepository(CloneDialogResult)` method which is indented to be used when the clone dialog returns a non-null `CloneDialogResult`.

While updating this I discovered that the create repository functionality is also calling `CloneRepository`. I this means that for every `NumberOfReposCreated` incremented, the `NumberOfClones` counter will be incremented as well. Oops! //cc @telliott27